### PR TITLE
scripts: profiler: RTT connection error handling

### PR DESCRIPTION
**Pull request by MarekPieta**
Add connection error handling for RTT host.

JIRA:DESK-395

### CODE REVIEW

**Comment by MarekPieta** [comment]
@pdunaj, @Qbicz, could you review?

File: scripts/profiler/rtt_nordic_profiler_host.py [hunk 1]
```diff
@@ -75,21 +75,39 @@ def connect(self):
         time.sleep(1)  # time required for initialization
         self.logger.info("Connected to device via RTT")
 
+    def shutdown(self):
+        self.disconnect()
+        self._read_remaining_events()
+        if self.event_filename is not None and self.event_types_filename is not None:
```
Comment (Qbicz) [inline scripts/profiler/rtt_nordic_profiler_host.py]: Should it try to write to a file called `''`?
Using `if self.event_filename and self.event_types_filename:` as a condition will both protect you from empty strings and look more readable.

Comment (MarekPieta) [reply scripts/profiler/rtt_nordic_profiler_host.py]: done

**Comment by MarekPieta** [comment]
I made requested changes

**Pull request merged**